index.js

```diff
@@ -36,7 +40,8 @@ function Router (opts) {
   this.ignoreTrailingSlash = opts.ignoreTrailingSlash || false
   this.maxParamLength = opts.maxParamLength || 100
   this.allowUnsafeRegex = opts.allowUnsafeRegex || false
-  this.tree = new Node()
+  this.versioning = opts.versioning || versioningStrategies.ACCEPT_VERSION
+  this.tree = new Node(undefined, undefined, undefined, undefined, undefined, this.versioning.storage())
```

Comment:
`Node` should be refactored to accept a single `options` object, there are too many parameters and the resulting code is quite confusing.

Reply:
agreed, have done it